extensions/browser/brave_extension_provider.cc

```diff
@@ -123,7 +123,12 @@ bool BraveExtensionProvider::MustRemainDisabled(
     const Extension* extension,
     disable_reason::DisableReason* reason,
     base::string16* error) const {
-  return extension->id() == brave_extension_id;
+  if (extension->id() == brave_extension_id)
+    return false;
+
+  if (reason)
+    *reason = disable_reason::DISABLE_BLOCKED_BY_POLICY;
```

Comment:
If disabled reason is not set for disabled extension, below DCHECK catches it.
```
void ExtensionRegistrar::DisableExtension(const ExtensionId& extension_id,
                                          int disable_reasons) {
  DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
  DCHECK_NE(disable_reason::DISABLE_NONE, disable_reasons);
```

Reply:
only if we return true though right? I think we can just always return false

Reply:
Always returns false seems fine.
So, removed some methods that are identical with parent methods.

It seems  that this PR has some conflicts(ex, GetTaskRunner()) with your working branches(manifest-handler and pdfjs branches).
Do you want me to wait landing your PR?